client/express-checkout/blocks/components/express-checkout-preview.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { Elements, ExpressCheckoutElement } from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InlineNotice from 'components/inline-notice';
+
+export const ExpressCheckoutPreviewComponent = ( {
+	stripe,
+	buttonType,
+	theme,
+	height,
+} ) => {
+	const [ canRenderButtons, setCanRenderButtons ] = useState( true );
+
+	const options = {
+		mode: 'payment',
+		amount: 1000,
+		currency: 'usd',
+	};
+
+	const mapThemeConfigToButtonTheme = ( paymentMethod, buttonTheme ) => {
+		switch ( buttonTheme ) {
+			case 'dark':
+				return 'black';
+			case 'light':
+				return 'white';
+			case 'light-outline':
+				if ( paymentMethod === 'googlePay' ) {
+					return 'white';
+				}
+
+				return 'white-outline';
+			default:
+				return 'black';
+		}
+	};
+
+	const googlePayType = buttonType === 'default' ? 'plain' : buttonType;
+
+	const applePayType = buttonType === 'default' ? 'plain' : buttonType;
+
+	let shouldOnlyShowApplePay = false;
+	if ( window.ApplePaySession && window.ApplePaySession.canMakePayments() ) {
+		shouldOnlyShowApplePay = true;
+	}
+
+	const buttonOptions = {
+		buttonHeight: Math.min( Math.max( height, 40 ), 55 ),
+		buttonTheme: {
+			googlePay: mapThemeConfigToButtonTheme( 'googlePay', theme ),
+			applePay: mapThemeConfigToButtonTheme( 'applePay', theme ),
+		},
+		buttonType: {
+			googlePay: googlePayType,
+			applePay: applePayType,
+		},
+		paymentMethods: {
+			link: 'never',
+			googlePay: shouldOnlyShowApplePay ? 'never' : 'always',
+			applePay: shouldOnlyShowApplePay ? 'always' : 'never',
+		},
+		layout: { maxColumns: 1, overflow: 'never' },
+	};
+
+	const onReady = ( { availablePaymentMethods } ) => {
+		if ( availablePaymentMethods ) {
+			setCanRenderButtons( true );
+		} else {
+			setCanRenderButtons( false );
+		}
+	};
+
+	if ( canRenderButtons ) {
+		return (
+			<div
+				key={ `${ buttonType }-${ height }-${ theme }` }
+				style={ { minHeight: `${ height }px`, width: '100%' } }
+			>
+				<Elements stripe={ stripe } options={ options }>
+					<ExpressCheckoutElement
+						options={ buttonOptions }
+						onClick={ () => {} }
+						onReady={ onReady }
+					/>
+				</Elements>
+			</div>
+		);
+	}
+
+	return (
+		<InlineNotice icon status="error" isDismissible={ false }>
+			{ __(
+				'Failed to preview the Apple Pay or Google Pay button. ' +
+					'Please ensure your store is served over HTTPS on a domain available to the public internet, ' +
+					'your device is configured to use Apple Pay or Google Pay, ' +
+					'and view this page using the Safari or Chrome browsers.',
+				'woocommerce-payments'
+			) }
```

Comment:
@elizaan36 @nikkivias @pierorocca In addition to the questions in the PR description, also looking for input on the info notice here. I think it's quite verbose but couldn't find any better, simpler, or more succinct phrasing that gets the point across.

Maybe we should add this to the documentation on PRBs and link to that instead?

Reply:
Same comment as [here](https://github.com/Automattic/woocommerce-payments/pull/9028#discussion_r1669604122) I think.

Reply:
Updated in 4863d3315de40a437190029d5b561d50ed2ff30a.